File: BOJ_1388/BOJ_1388.c
#include <stdio.h>
#include <string.h>
#include <stdlib.h>
int main(){
        int N, M, HWOOD=0, VWOOD =0, SEQUENCE=0;
        char FLOOR[64][64];
        char FLAT = '-';
        char VERT = '|';
        scanf("%d %d", &N, &M);
        for(int i=0; i<N; i++){
                scanf("%s", FLOOR[i]);
        }
        for(int i=0; i<N; i++){
                for(int j=0; j<M; j++){
                        if(FLOOR[i][j] == FLAT) SEQUENCE = 1;
                        else if(FLOOR[i][j] == VERT && SEQUENCE == 1){
                                HWOOD++;
                                SEQUENCE = 0;
                        }
                        else continue;
                }
                HWOOD = HWOOD + SEQUENCE;
                SEQUENCE = 0;
        }
        SEQUENCE = 0;
        for(int i=0; i<M; i++){
                for(int j=0; j<N; j++){
                        if(FLOOR[j][i] == VERT) SEQUENCE = 1;
                        else if(FLOOR[j][i] == FLAT && SEQUENCE == 1){
                                VWOOD++;
                                SEQUENCE = 0;
                        }
                        else continue;
                }
                VWOOD = VWOOD + SEQUENCE;
                SEQUENCE = 0;
        }
        printf("%d\n", HWOOD+VWOOD);

        return 0;
}

File: BOJ_1388/BOJ_1388.java
import java.io.*;

class Main{
        public static void main(String[] args)
                        throws IOException{
                        BufferedReader BR = new BufferedReader(new InputStreamReader(System.in));
                        BufferedWriter BW = new BufferedWriter(new OutputStreamWriter(System.out));
                        StringBuilder SB = new StringBuilder();
                        String[] NUM = BR.readLine().split(" ");
                        String[][] FLOOR = new String[100][100];
                        int N = Integer.parseInt(NUM[0]);
                        int M = Integer.parseInt(NUM[1]);
                        int VWOOD =0;
                        int HWOOD =0;
                        int SEQUENCE = 0;
                        for(int i=0; i<N; i++){
                                FLOOR[i] = BR.readLine().split("");

                        }
                        for(int i=0; i<N; i++){
                                for(int j=0; j<M; j++){
                                        if(FLOOR[i][j].equals("-") == true) SEQUENCE = 1;
                                        else if(FLOOR[i][j].equals("|") == true && SEQUENCE == 1){
                                                HWOOD++;
                                                SEQUENCE = 0;
                                        }
                                        else continue;
                                }
                                HWOOD = HWOOD + SEQUENCE;
                                SEQUENCE = 0;
                        }
                        SEQUENCE = 0;
                        for(int i=0; i<M; i++){
                                for(int j=0; j<N; j++){
                                        if(FLOOR[j][i].equals("|") == true) SEQUENCE = 1;
                                        else if(FLOOR[j][i].equals("-") == true && SEQUENCE == 1){
                                                VWOOD++;
                                                SEQUENCE = 0;
                                        }
                                        else continue;
                                }
                                VWOOD = VWOOD + SEQUENCE;
                                SEQUENCE = 0;
                        }
                        SB.append(VWOOD+HWOOD);
                        BW.write(SB+"\n");
                        BW.flush();

        }
}
